src/modules/data/audioStream.c

```diff
@@ -39,25 +91,71 @@ size_t lovrAudioStreamDecode(AudioStream* stream, int16_t* destination, size_t s
   size_t samples = 0;
 
   while (samples < capacity) {
-    int count = stb_vorbis_get_samples_short_interleaved(decoder, channelCount, buffer + samples, (int) (capacity - samples));
+    size_t count = 0;
+    if (decoder) {
+      count = stb_vorbis_get_samples_short_interleaved(decoder, channelCount, buffer + samples, (int)(capacity - samples));
+    } else {
+      count = dequeue_raw(stream, buffer + samples, (int)(capacity - samples));
+      stream->queueLengthInSamples -= count;
+    }
     if (count == 0) break;
     samples += count * channelCount;
   }
 
   return samples;
 }
 
+bool lovrAudioStreamAppendRawBlob(AudioStream* stream, struct Blob* blob)
+{
+  lovrAssert(lovrAudioStreamIsRaw(stream), "Raw PCM data can only be appended to a raw AudioStream (see constructor that takes channel count and sample rate)")
+  if (stream->queueLimitInSamples != 0 && stream->queueLengthInSamples + blob->size/sizeof(int16_t) >= stream->queueLimitInSamples) {
+    return false;
+  }
+  lovrRetain(blob);
+  arr_push(&stream->queuedRawBuffers, blob);
+  stream->queueLengthInSamples += blob->size / sizeof(int16_t);
+  return true;
+}
+
+bool lovrAudioStreamAppendRawSound(AudioStream* stream, struct SoundData* sound)
+{
+  lovrAssert(sound->channelCount == stream->channelCount && sound->bitDepth == stream->bitDepth && sound->sampleRate == stream->sampleRate, "SoundData and AudioStream formats must match");
+  lovrAudioStreamAppendRawBlob(stream, &sound->blob);
+  return true;
+}
+
+size_t lovrAudioStreamGetQueueLength(AudioStream* stream)
+{
+  lovrAssert(lovrAudioStreamIsRaw(stream), "Queue length is only available on raw streams");
+  return stream->queueLengthInSamples;
+}
+
+bool lovrAudioStreamIsRaw(AudioStream* stream)
+{
+  return stream->decoder == NULL;
+}
```

Comment:
Could the new functions here use the same bracing convention as the other functions?  e.g.

```c
bool lovrAudioStreamIsRaw(AudioStream* stream) {
  return stream->decoder == NULL;
}

Reply:
yes of course! Sorry, new line for bracket is the alloverse coding convention, I got mixed up :) you can definitely nitpick on style no worries, I'll fix!